main_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"syscall"
+	"testing"
+	"time"
+
+	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+
+	"github.com/firecracker-microvm/firecracker-go-sdk"
+	client "github.com/firecracker-microvm/firecracker-go-sdk/client"
+	ops "github.com/firecracker-microvm/firecracker-go-sdk/client/operations"
+	"github.com/go-openapi/strfmt"
+	"github.com/sirupsen/logrus"
+)
+
+var kernelImagePath = "./vmlinux"
+
+const rootDrivePath = "/dev/null"
+
+func init() {
+	if v := os.Getenv("KERNELIMAGE"); len(v) != 0 {
+		kernelImagePath = v
+	}
+}
+
+func TestFireCTL(t *testing.T) {
+	const firectlName = "./firectl"
+	if _, err := os.Stat(firectlName); os.IsNotExist(err) {
+		t.Skip("Missing firectl binary")
```

Comment:
As above, I'd prefer that this is `t.Fail` unless the user explicitly skips the test.